walkeeper/src/replication.rs

```diff
@@ -79,10 +88,13 @@ impl ReplicationConn {
 
     /// Handle incoming messages from the network.
     /// This is spawned into the background by `handle_start_replication`.
-    fn background_thread(
-        mut stream_in: impl Read,
-        subscriber: impl HsFeedbackSubscriber,
-    ) -> Result<()> {
+    fn background_thread(mut stream_in: impl Read, timeline: Arc<Timeline>) -> Result<()> {
+        let mut state = ReplicaState::new();
+        let replica = timeline.add_replica();
```

Comment:
It looks like race condition is possible. Two calls to `background_thread` can end up with the same value of replica id, because we do not mark the slot as allocated both invocations can pick the same index of first vacant slot